### ISSUE TITLE
adjust_supply refactoring in unit tests

### ISSUE THREAD

**Issue by theoreticalbts** [steemit/steem#1992]
I approved PR #1983, but there are some rough edges.  We want to replace:

```
adjust_balance( alice, amount );
adjust_supply( amount );
```

and instead use:

```
fund( alice, amount );
```

So essentially, we need to add supporting code to `fund()` for SMT's.

We should also simplify the `fund()` method body to use `adjust_balance()` and `adjust_supply()` where possible, it currently re-implements (substantial parts of) `adjust_balance()` and `adjust_supply()`.

**Comment by Kiwonik** [comment]
@youkaicountry I added the issue to SMT project triage, as I found it necessary there.

**Comment by Kiwonik** [comment]
I could not find easy way to simplify `fund()` body in part handling legacy asset types (STEEM/SBD). I extended it with SMT handling and replaced the code in tests with `fund` calls.
I also added a new macro `FUND` that encapsulates `fund` & `generate_block` calls to facilitate writing tests without problems related to lack of following `generate_block`.
BTW I think `ACTORS` macro should be extended with a call to `generate_block` for the same reason, and a similar `CONVERT` macro should be created.
